### PR TITLE
Change  default correlate_off to True for ExcessMapEstimator

### PR DESCRIPTION
**Pull request by registerrier**
**Description**

This pull request reverts the change to have by default `correlate_off=False` for `ExcessMApEstimator`. As noted by @AtreyeeS , this leads to poor results when using stacked datasets.
The main situation where this makes sense is when the `ExcessMapEstimator` is applied on  non stacked datasets. We can therefore leave it as an option but change its default.

**Dear reviewer**

### CODE REVIEW

**Comment by mdebony** [comment]
Hi,
I don't find the original conversation about the merge request. Would it be possible to have a link to it ?

The reason this parameter was set by default to False, is than it's expected the MapDatasetOnOff is generated by a RingBackgroundMaker. As the ring background estimation technique already correlate OFF, correlate again OFF after the ring background cause than some OFF events will be counts multiple-time. In that case, the OFF count will be too high which can lead to an over estimation of the significance.

**Comment by registerrier** [comment]
> Hi,
> I don't find the original conversation about the merge request. Would it be possible to have a link to it ?
>
Yes it is this one : #3140 

> The reason this parameter was set by default to False, is than it's expected the MapDatasetOnOff is generated by a RingBackgroundMaker. As the ring background estimation technique already correlate OFF, correlate again OFF after the ring background cause than some OFF events will be counts multiple-time. In that case, the OFF count will be too high which can lead to an over estimation of the significance.

 Indeed, you are right. The issue is that one cannot apply it this way when stacking observations because of the various offset cuts. This was noticed by @AtreyeeS . I will paste some example showing the results.

The correct approach in that case should be to keep individual datasets, apply the `ExcessMapEstimator` on each of them and stack the result.
So we still need the functionality not to correlate, but it needs to be properly documented how to do it. And at some point we will need to allow for multiple datasets to be be passed.

**Comment by adonath** [comment]
@registerrier and @AtreyeeS Do we have an example (e.g. in the docs) where the current default leads to a poor result?

**Comment by AtreyeeS** [comment]
Do we want to show ugly results in our docs?

**Comment by bkhelifi** [comment]
In the docstring, there is not yet the recommandations of use of the correlation. Should we add some doc?

**Comment by adonath** [comment]
> Do we want to show ugly results in our docs?

No not at all, I was just asking for a visual example for myself to see the effect once...

**Comment by adonath** [comment]
Maybe you can share a screenshot as well, if there is no example in our docs...

**Comment by AtreyeeS** [comment]
Here is a minimal example that @registerrier  recreated - significance map from simulated wobble observations with a ring background. The edge effects are quite clear.

unknown-8

**Comment by adonath** [comment]
Thanks for the illustration @AtreyeeS! If I understood correctly there is no easy way to make this correct, right? 
- Either we correlate both `counts` and `counts_off` for the stacked case, but then the off counts are twice correlated
- If we don't correlate the `counts_off` again we end up with the artefacts as in @AtreyeeS example
- The only way to make it correct is correlate on the dataset level and stack the datasets again, which is a whole new algorithm

To improve the code structure a bit I was wondering, whether we should introduce `MapDataset.convolve()` and `MapDatasetOnOff.convolve()`?

**Comment by registerrier** [comment]
> Thanks for the illustration @AtreyeeS! If I understood correctly there is no easy way to make this correct, right?
> 
> * Either we correlate both `counts` and `counts_off` for the stacked case, but then the off counts are twice correlated
> * If we don't correlate the `counts_off` again we end up with the artefacts as in @AtreyeeS example
> * The only way to make it correct is correlate on the dataset level and stack the datasets again, which is a whole new algorithm

 That's a good summary. We can possibly show how to do the observation based excess map calculation and stacking in, e.g. a recipe, but this requires some adaptation of the `ExcessMapEstimator` to be done properly as it is necessary to return correlated acceptances as well. This is not the case now.

> To improve the code structure a bit I was wondering, whether we should introduce `MapDataset.convolve()` and `MapDatasetOnOff.convolve()`?

I am not sure I see the exact use case here. We could create an internal method/function to perform correlation of the various quantities stored in the dataset but I don't really see what would be a convolved dataset.

**Comment by adonath** [comment]
> I am not sure I see the exact use case here. We could create an internal method/function to perform correlation of the various quantities stored in the dataset but I don't really see what would be a convolved dataset.

I remember one other use case, where @QRemy extracted probably O(100) spectral datasets of the same size and found that the correlation beforehand sped up the process a lot. But I agree it's a rare use-case and must be taken with care...

**Comment by adonath** [comment]
Just for the record: one could provide the a static method like `ExcessMapEstimator.correlate_dataset(dataset, kernel)`...

File: gammapy/estimators/excess_map.py [hunk 1]
```diff
@@ -67,10 +67,6 @@ class ExcessMapEstimator(Estimator):
     If a model is set on the dataset the excess map estimator will compute the excess taking into account
     the predicted counts of the model.
 
-    Some background estimation techniques like ring background or adaptive ring background will provide already
```
Comment (adonath) [inline gammapy/estimators/excess_map.py:70]: Replace with a statement / warning on the fact that it is correlated twice? Whatever this does to the statistics...

**Comment by adonath** [comment]
Again for the record: will show the "correct" way of computing the map in a Gammapy "recipe".

**Comment by adonath** [comment]
@registerrier I think the remaining action item here is to extend the documentation of the `ExcessMapEstimator`?

**Comment by adonath** [comment]
@registerrier and @AtreyeeS I added a note to the docs of the `ExcessMapEstimator` see https://github.com/gammapy/gammapy/pull/3383/files#diff-d51c4d43972776ddc24f7afb00b2dff45134553ab42df8cb573688e25d5682b4R70 

Do you think the will be sufficient?

**Comment by AtreyeeS** [comment]
> I added a note to the docs of the `ExcessMapEstimator` see https://github.com/gammapy/gammapy/pull/3383/files#diff-d51c4d43972776ddc24f7afb00b2dff45134553ab42df8cb573688e25d5682b4R70
> 
> Do you think the will be sufficient?

Thanks @adonath ! This looks clear to me. 
I wonder if we should change our `RingBackground` tutorial to do this accordingly?

**Pull request merged**